src/Confluent.Kafka/Consumer.cs

```diff
@@ -824,12 +824,11 @@ public Consumer(IEnumerable<KeyValuePair<string, object>> config)
             if (defaultTopicConfig != null)
             {
                 var topicConfigHandle = SafeTopicConfigHandle.Create();
-                if (config != null)
-                {
-                    defaultTopicConfig.ToList().ForEach((kvp) => { topicConfigHandle.Set(kvp.Key, kvp.Value.ToString()); });
-                }
+                defaultTopicConfig.ToList().ForEach(
+                    (kvp) => { topicConfigHandle.Set(kvp.Key, kvp.Value.ToString()); }
+                );
                 LibRdKafka.conf_set_default_topic_conf(configPtr, topicConfigHandle.DangerousGetHandle());
```

Comment:
This will overwrite any topic configurations set by fallthru on the main config, i.e.,  this will overwrite the implicitly created default topic config (if any).
I think it is better to set the key-vals from defaultTopicConfig directly on the configPtr instead.